{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "07c2125e",
   "metadata": {},
   "outputs": [],
   "source": [
    "from tensorflow import keras\n",
    "from tensorflow.keras.models import Sequential\n",
    "from tensorflow.keras.layers import Dense, Flatten, Activation\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "503c628b",
   "metadata": {},
   "source": [
    "#### Potential imporvements to model\n",
    "* tune with hyperparameters\n",
    "* try different loss and optimizers as well as different activation\n",
    "    * potential loss functions\n",
    "        * BinaryCrossentropy\n",
    "        * categorical_crossentropy\n",
    "        * sparse_categorical_crossentropy\n",
    "        * poisson function\n",
    "    * potential activation functions\n",
    "        * relu\n",
    "        * leaky relu\n",
    "        * tanh (maybe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f5c708c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "def build_model():\n",
    "    model = Sequential()\n",
    "    model.add(Dense(5, input_dim=5, activation='relu'))\n",
    "    model.add(Dense(4, activation='relu'))\n",
    "    model.add(Dense(1, activation='sigmoid'))\n",
    "    model.compile(loss='binary_crossentropy', optimizer='adam', metrics=[\"accuracy\"])\n",
    "    \n",
    "    return model\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "fd697297",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-11-30 12:36:03.189638: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory\n",
      "2021-11-30 12:36:03.189665: W tensorflow/stream_executor/cuda/cuda_driver.cc:312] failed call to cuInit: UNKNOWN ERROR (303)\n",
      "2021-11-30 12:36:03.189682: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (theBigFatPanda): /proc/driver/nvidia/version does not exist\n",
      "2021-11-30 12:36:03.189888: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN)to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA\n",
      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2021-11-30 12:36:03.215516: I tensorflow/core/platform/profile_utils/cpu_utils.cc:104] CPU Frequency: 1497600000 Hz\n",
      "2021-11-30 12:36:03.216331: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x54a46c0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:\n",
      "2021-11-30 12:36:03.216346: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version\n"
     ]
    }
   ],
   "source": [
    "model = build_model()\n",
    "\n",
    "epochs = 100\n",
    "model.fit(X_train, Y_train, epochs=epochs, verbose=0)\n",
    "\n",
    "\n",
    "Y_prediction = model.predict_proba(X_test)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74717c76",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
